pkg/utils/envoyutils/admincli/client.go

```diff
@@ -308,6 +310,44 @@ func (c *Client) GetSingleListenerFromDynamicListeners(
 	return &listener, nil
 }
 
+// GetSingleRouteConfig queries for a single, route configuration in the envoy config dump
+// and returns it as an envoy v3.RouteConfiguration.
+// This helper will only work if the provided name_regex matches a single route configuration
+// but will always use the first set of configs returned regardless
+func (c *Client) GetSingleRouteConfig(
+	ctx context.Context,
+	routeConfigNameRegex string,
+) (*route_configv3.RouteConfiguration, error) {
+	queryParams := map[string]string{
+		"resource":   "dynamic_route_configs",
+		"name_regex": routeConfigNameRegex,
+	}
+	cfgDump, err := c.GetConfigDump(ctx, queryParams)
+	if err != nil {
+		return nil, fmt.Errorf("could not get envoy config_dump from adminClient: %w", err)
+	}
+
+	configs := cfgDump.GetConfigs()
+
+	// if no dynamic route configs name matching routeConfigNameRegex or before envoy is full configured
```

Comment:
Mmmm... can this be a tad more English? 😁 
Perhaps...
```suggestion
	// if no dynamic route config's name matches routeConfigNameRegex or before envoy is fully configured
```